lib/matplotlib/ticker.py

```diff
@@ -654,30 +626,76 @@ def get_offset(self):
 
     def set_locs(self, locs):
         'set the locations of the ticks'
+        # Calculate automatic scaling factor / scientific notation
+        # and automatic offset value. This communicates with pprint_val()
+        # which subtracts the offset and scales the result
         self.locs = locs
 
         if len(self.locs) > 0:
             vmin, vmax = self.axis.get_view_interval()
-            d = abs(vmax - vmin)
-            self._set_offset(d)
-
-            self._set_orderOfMagnitude(d)
+            range = abs(vmax - vmin)
+
+            # Automatic offsets are needed if not disabled by rcParam
+            # nor if an offset has already been set
+            if(self.get_useAutomaticOffset() is True and
+               self.get_useOffset() is False):
+                autoOffset = self._calculateAutomaticOffset(range)
+                self.set_useOffset(autoOffset)
+
+            # Automatic scaling/scientific notation is needed if not
+            # already been set
+            if self.get_useScalingFactor() is False:
+                autoScalFac = self._calculateAutomaticScaling(range)
+                self.set_useScalingFactor(autoScalFac)
+
+            # Set the format string for the ticks
             self._set_format(vmin, vmax)
 
-    def _set_offset(self, range):
-        # Determine if an offset is needed and if so, set it.
-        # Don't do this if offset has already been set of it is
-        # rcParams forbids automatic offset
-        if(rcParams['axes.formatter.useoffset'] is False or
-                self.get_useOffset() is True):
-            return
+    def pprint_val(self, x):
+        # apply offset and scaling/scientific notation here
```

Comment:
The actual offset and scaling factor are used here to change the ticks. This is the same code as in master